src/Data/Array.purs

```diff
@@ -369,6 +370,13 @@ concatMap = flip bind
 -- | creating a new array.
 foreign import filter :: forall a. (a -> Boolean) -> Array a -> Array a
 
+-- | Partition an array using a predicate function, creating a set of
+-- | new arrays. One for the values satisfying the predicate function
+-- | and one for values that don't.
+foreign import partition :: forall a. (a -> Boolean)
+                         -> Array a
+                         -> { yes :: Array a, no :: Array a }
```

Comment:
I like the return type, I frequently have to consult Hackage to remember the right way round in Haskell.